### PR TITLE
Add walltime metrics on sampling

### LINKED ISSUE #72: Add walltime metrics on sampling
Sampling is the process of selecting a token with the highest relative probability from a series of logits

### PR DESCRIPTION
**Pull request by guojidan**
closed #72 

😕 Maybe we should handle it this way

### CODE REVIEW

**Comment by flaneur2020** [comment]
thank you for your contribution!

this pr adds the walltime metrics for the `add` primitive, we still need add the walltime metrics for the `sampling` phase, which locates in the llama2.rs file, so the #72 issue is still better to be kept open :)

**Comment by guojidan** [comment]
ok, I will continue work on it
